### PR TITLE
Reduce local query runner

### PR DESCRIPTION
**Pull request by dain**
## Description

LocalQueryRunner is complex and difficult to maintain.  More importantly, LocalQueryRunner is not a full server and simulates important parts of a query.  The PR works to replace LocalQueryRunner with StandaloneQueryRunner, and where that is not possible reduces the scope of the dependence on LocalQueryRunner extensions. 

The conversion to StandaloneQueryRunner fixes: #20043

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: core/trino-main/src/test/java/io/trino/operator/scalar/timestamp/TestExtract.java [hunk 1]
```diff
@@ -336,7 +335,6 @@ public void testSecond()
     public void testMillisecond()
     {
         assertThatThrownBy(assertions.expression("EXTRACT(MILLISECOND FROM TIMESTAMP '2020-05-10 12:34:56')")::evaluate)
```
[the remark this answers is not in the document]
Comment (electrum) [reply core/trino-main/src/test/java/io/trino/operator/scalar/timestamp/TestExtract.java:337]: All of these `ParsingException` conversions could do `.hasErrorCode(SYNTAX_ERROR)`

File: core/trino-main/src/test/java/io/trino/sql/query/TestRowFilter.java [hunk 2]
```diff
@@ -755,15 +757,13 @@ public void testRowFilterWithHiddenColumns()
                 .assertThat()
                 .skippingTypesCheck()
                 .matches("VALUES (BIGINT '0', 'ALGERIA', BIGINT '0', ' haggle. carefully final deposits detect slyly agai')");
-        assertThatThrownBy(() -> assertions.query("INSERT INTO mock.tiny.nation_with_hidden_column VALUES (101, 'POLAND', 0, 'No comment')"))
-                .isInstanceOf(TrinoException.class)
+        assertTrinoExceptionThrownBy(() -> assertions.query("INSERT INTO mock.tiny.nation_with_hidden_column VALUES (101, 'POLAND', 0, 'No comment')"))
```
Comment (electrum) [inline core/trino-main/src/test/java/io/trino/sql/query/TestRowFilter.java:760]: Assert error code for all of these?

File: core/trino-main/src/test/java/io/trino/sql/query/TestShowQueries.java [hunk 3]
```diff
@@ -39,31 +41,29 @@ public class TestShowQueries
...
-                                                .setName("colabc")
-                                                .setType(BIGINT)
-                                                .build()))
-                        .withListSchemaNames(session -> ImmutableList.of("mockschema"))
-                        .withListTables((session, schemaName) -> ImmutableList.of("mockTable"))
-                        .build(),
-                ImmutableMap.of());
+        queryRunner.installPlugin(new MockConnectorPlugin(MockConnectorFactory.builder()
+                .withGetColumns(schemaTableName ->
+                        ImmutableList.of(
```
Comment (electrum) [inline core/trino-main/src/test/java/io/trino/sql/query/TestShowQueries.java:50]: Nit: could put on previous line

File: core/trino-main/src/test/java/io/trino/type/TestArrayOperators.java [hunk 4]
```diff
@@ -265,11 +265,8 @@ public void testArrayToArrayCast()
     public void testArraySize()
     {
         int size = toIntExact(MAX_FUNCTION_MEMORY.toBytes() + 1);
-        assertTrinoExceptionThrownBy(() -> assertions.expression("array_distinct(ARRAY['" +
-                                                                 "x".repeat(size) + "', '" +
-                                                                 "y".repeat(size) + "', '" +
-                                                                 "z".repeat(size) +
-                                                                 "'])").evaluate())
+        assertTrinoExceptionThrownBy(
+                () -> assertions.expression("array_distinct(ARRAY[lpad('', %s, 'x'), lpad('', %s, 'y'), lpad('', %s, 'z')])".formatted(size, size, size)).evaluate())
```
Comment (electrum) [inline core/trino-main/src/test/java/io/trino/type/TestArrayOperators.java:269]: You can use `%1$s` to avoid repeating the argument

**Review by electrum — APPROVED** [review]
Looks good, other than the client changes

**Comment by mosabua** [comment]
👋 @dain  .. I assume you will continue to run with this with @electrum and get it to a merge.

**Pull request merged**